### PR TITLE
[Relay][Frontend][Onnx] Add RNN operation for ONNX frontend

### PR DESCRIPTION
**Pull request by zhang-yi-chi**
Add RNN operator for ONNX frontend. Note that currently onnxruntime (<= 1.12.0) has wrong default values (all zeros) for all activations so I removed   default value tests for HardSigmoid.

cc @mbrookhart @AndrewZhaoLuo

### CODE REVIEW

**Comment by AndrewZhaoLuo** [comment]
I'll take a look today

File: python/tvm/relay/frontend/onnx.py [hunk 1]
```diff
@@ -2756,6 +2757,138 @@ def _activation_needs_beta(cls, activation):
...
+            raise ValueError("num_directions must be either 1 or 2!")
+
+        X_shape = infer_shape(X)
+        hidden_size = infer_shape(Rp)[-1]
+        batch_size = X_shape[1]
+
+        if Hp_0 is None:
+            Hp_0 = _op.zeros((num_directions, batch_size, hidden_size), W_dtype)
+
+        if "activations" in attr:
```
Comment (AndrewZhaoLuo) [inline python/tvm/relay/frontend/onnx.py:2819]: Can you refactor this as it seems identical to what's in LSTM?

**Review by AndrewZhaoLuo — COMMENTED** [review]
I think it looks fine. My main concern is about the high amount of similar code between this and LSTM. Can you try a refactor to share relevant parts with each other instead of copy paste?

Also q about tests

File: tests/python/frontend/onnx/test_forward.py [hunk 2]
```diff
@@ -5212,7 +5341,7 @@ def verify_eyelike(indata, dynamic=False):
     "test_reduce_sum_keepdims_random",
     "test_reduce_sum_negative_axes_keepdims_example",
     "test_reduce_sum_negative_axes_keepdims_random",
-    "test_rnn_seq_length",
+    "test_rnn_batchwise",
```
[the remark this answers is not in the document]
Comment (zhang-yi-chi) [reply tests/python/frontend/onnx/test_forward.py]: I found that ONNX doesn't has test_rnn_batchwise node so I simply removed this line in the following commit.

File: python/tvm/relay/frontend/onnx.py [hunk 3]
```diff
@@ -2756,6 +2757,138 @@ def _activation_needs_beta(cls, activation):
...
+                _op.stack([forward_outputs[i], reverse_outputs[seq_len - 1 - i]], axis=0)
+            )
+
+        return (
+            _op.stack(final_outputs, axis=0),
+            _op.stack([fw_H_t, rev_H_t], axis=0),
+        )
+
+    @classmethod
+    def _impl_v7(cls, inputs, attr, params):
```
[the remark this answers is not in the document]
Comment (zhang-yi-chi) [reply python/tvm/relay/frontend/onnx.py]: Sure. I found that opset 14 has an extra attr `layout` but onnxruntime doesn't support `layout == 1`, so I commented out the test for `layout == 1` for now.

File: python/tvm/relay/frontend/onnx.py (hunk 1 above)
Comment (zhang-yi-chi) [reply python/tvm/relay/frontend/onnx.py:2804]: I refactored both onnx.py and test_forward.py.

**Review by AndrewZhaoLuo — APPROVED** [review]
LGTM, thanks!

**Pull request merged**